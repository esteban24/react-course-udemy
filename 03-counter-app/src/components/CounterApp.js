import React, { useState } from 'react'
import PropTypes from 'prop-types';
import App from '../App';


const CounterApp = ({ value = 10 }) => {

    const [ counter, setCounter ] = useState( value ); // return [value1, value2]

    const handleAdd = (e) => {
        setCounter( counter + 1 );
        // setCounter((counter) => counter + 1);
    }

    const handleSubstract = (e) => {
        setCounter( counter - 1 );
        // setCounter((counter) => counter - 1);
    }

    const handleReset = (e) => {
        // setCounter( counter + 1 );
        setCounter(value);
    }

    return (
        <>
            <h1>Counter App</h1>
            <h2>{ counter }</h2>
            <button onClick={ handleAdd }>+</button>
            <button onClick={ handleSubstract }>-</button>
            <button onClick={ handleReset }>Reset</button>
        </>
    )
}

App.propTypes = {
    value: PropTypes.number
}

export default CounterApp;

