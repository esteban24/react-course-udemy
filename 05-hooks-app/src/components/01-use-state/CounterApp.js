import React, { useState } from 'react'
import './counter.css';

export const CounterApp = () => {
    const [ state, setState] = useState({
        counterPrimary: 1,
        counterSecundary: 10,
    });

    const count = () => {
        setState({ ...state, counterPrimary: state.counterPrimary + 1 })
    }

    return (
        <>
            <h1>Counter Primary { state.counterPrimary }</h1>
            <h1>Counter Secundary { state.counterSecundary }</h1>
            <hr />
            <button onClick={ count } className="btn btn-primary"> +1 </button>
        </>
    )
}
