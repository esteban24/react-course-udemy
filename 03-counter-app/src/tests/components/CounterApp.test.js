import React from 'react';
import '@testing-library/jest-dom';
import { describe, expect, test, beforeEach } from '@jest/globals';
import { shallow } from 'enzyme';

import CounterApp from '../../components/CounterApp';

describe('CounterApp.js test suite', () => {
    let counterApp = shallow(<CounterApp />);

    beforeEach( () => {
        counterApp = shallow(<CounterApp />);
    });

    test('should match <CounterApp /> snapshot', () => {
        expect(counterApp).toMatchSnapshot();
    });

    test('should return default value = 100', () => {
        const defaultValue = 100;
        const counterApp = shallow(<CounterApp value={ defaultValue }/>);

        const actualValue = counterApp.find('h2').text().trim();
        expect(Number.parseInt(actualValue)).toBe(defaultValue);
    });

    test('should increment with button +', () => {
        const plusButton = counterApp.find('button').at(0);

        plusButton.simulate('click');
        const actualValue = counterApp.find('h2').text().trim();

        expect(actualValue).toBe("11");
    });

    test('should decrement with button -', () => {
        const plusButton = counterApp.find('button').at(1);

        plusButton.simulate('click');
        const actualValue = counterApp.find('h2').text().trim();

        expect(actualValue).toBe("9");
    });

    test('should reset value to default with button reset', () => {
        const defaultValue = 105;
        const counterApp = shallow(<CounterApp value={ defaultValue }/>);

        // Increment to change value and then check if value is incremented
        const plusButton = counterApp.find('button').at(0);
        plusButton.simulate('click');
        expect(counterApp.find('h2').text().trim()).toBe("106");

        // Check if reset actually works
        const resetButton = counterApp.find('button').at(2);
        resetButton.simulate('click');
        expect(counterApp.find('h2').text().trim()).toBe("105");
    })

});
