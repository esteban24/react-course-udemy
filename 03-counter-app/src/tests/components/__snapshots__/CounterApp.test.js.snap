// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterApp.js test suite should match <CounterApp /> snapshot 1`] = `
<Fragment>
  <h1>
    Counter App
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +
  </button>
  <button
    onClick={[Function]}
  >
    -
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
</Fragment>
`;
